koji_containerbuild/plugins/builder_containerbuild.py

```diff
@@ -543,10 +543,12 @@ def handler(self, src, target_info, arch, output_template, scratch=False,
         files = []
         if response.is_succeeded() and response.get_tar_metadata_filename():
             sourceimage = response.get_tar_metadata_filename()
-            if sourceimage.endswith(".tar.gz"):
-                targetimage = "image-%s.tar.gz" % arch
-            else:
-                targetimage = "image-%s.tar" % arch
+            targetimage = "image-%s.tar" % arch
+
+            extension = self.osbs().get_compression_extension()
+            if extension:
+                targetimage = targetimage + extension
```

Comment:
Use string formatting here instead as this can easily fail if we got unicode from osbs and try to append it to str.